src/Build/Logging/BaseConsoleLogger.cs

```diff
@@ -329,6 +329,33 @@ internal void IsRunningWithCharacterFileType()
         /// </summary>
         internal bool IsVerbosityAtLeast(LoggerVerbosity checkVerbosity) => Verbosity >= checkVerbosity;
 
+        /// <summary>
+        /// Returns the minimum logger verbosity required to log a message with the given importance.
+        /// </summary>
+        /// <param name="importance">The message importance.</param>
+        /// <param name="lightenText">True if the message should be rendered using lighter colored text.</param>
+        /// <returns>The logger verbosity required to log a message of the given <paramref name="importance"/>.</returns>
+        internal static LoggerVerbosity ImportanceToMinimumVerbosity(MessageImportance importance, out bool lightenText)
```

Comment:
This would best reside somewhere else (LoggingService?) if we were to support third party loggers, right?

Reply:
I wonder if supporting 3rd party loggers wouldn't require extending the contract between MSBuild and loggers where the logger would indicate its verbosity directly in terms of `MessageImportance`. Just like they choose if they're interested in, say, `BuildStarted` and `BuildFinished` today, they would also say they want messages only with the given importance.

It's true that we may want to expose a helper with a default mapping between verbosity and importance but ultimately it is up to the logger what exactly it chooses to log.